packages/lesswrong/components/editor/LastEditedInWarning.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib/components';
+import { editorTypeToDisplay, EditorTypeString, EditorContents, EditorChangeEvent } from './Editor';
+import { useConvertDocument } from './useConvertDocument';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  lastEditedWarning: {
+    color: theme.palette.error.main,
+  },
+  clickHereColor: {
+    color: theme.palette.primary.main
+  },
+});
+
+const LastEditedInWarning = ({initialType, currentType, defaultType, value, setValue, classes}: {
+  initialType: EditorTypeString,
+  currentType: EditorTypeString,
+  defaultType: EditorTypeString,
+  value: EditorContents,
+  setValue: (change: EditorChangeEvent)=>void,
+  classes: ClassesType
+}) => {
+  const { Loading, Typography } = Components;
+  const {convertDocument, loading, error} = useConvertDocument({
+    onCompleted: (result: EditorContents) => {
+      setValue({
+        contents: result,
+        autosave: false,
+      });
+    }
+  });
+  
+  return <div>
+    {loading && <Loading/>}
+    <Typography variant="body2" className={classes.lastEditedWarning}>
+      This document was last edited in {editorTypeToDisplay[initialType].name} format. Showing the{' '}
+      {editorTypeToDisplay[currentType].name} editor.{' '}
+      <a
+        className={classes.clickHereColor}
+        onClick={() => {
+          convertDocument(value, 'ckEditorMarkup');
+        }}
+      >
+        Click here
+      </a>
+      {' '}to switch to the {editorTypeToDisplay['ckEditorMarkup'].name} editor (the default editor).
```

Comment:
These changes were made in response to various bugs that arose after we enabled ckEditor collab

The change was very intentional. Conversion between document types is complicated, bug-prone, and moving away from ckEditor disables sharing – so we decided to discourage (or at least not encourage) the use of anything except LWDocs, while still providing some support for Markdown which seems warranted (something like 15% of posts/users use Markdown, though perhaps mostly for importing).

Reply:
If 15% of users use markdown, do you really want to put giant red text at the bottom of all their posts and comments?

Fake edit: Ok, well you haven't actually changed the conditions upon which you show this element. So the good news is you won't show it on the bottom of all the markdown users markdown posts, but the bad news is you will show this on the bottom of the markdown users ckeditor posts, which seems pretty pointless dontcha think?

Fake edit 2: Actually I can't ever get this element to show.

Reply:
Robert and I looked into this. I *think* I agree with JP here, but want to check in with Ruby about anything I might have been forgetting.

Robert and I changed it so that it doesn't show the message for markdown users on markdown posts – it only shows it on markdown posts for user's who's default editor is currently `ckEditorMarkup`

Reply:
@darkruby501